runtime-light/stdlib/server/args-functions.cpp

```diff
@@ -0,0 +1,90 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2025 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include <cctype>
+#include <cstdint>
+#include <string_view>
+
+#include "runtime-common/core/runtime-core.h"
+#include "runtime-light/state/instance-state.h"
+#include "runtime-light/stdlib/server/args-functions.h"
+
+Optional<array<mixed>> f$getopt(const string &short_options, const array<string> &long_options,
+                                [[maybe_unused]] Optional<std::optional<std::reference_wrapper<string>>> rest_index) noexcept {
+  if (const auto &instance_st{InstanceState::get()}; instance_st.image_kind() != ImageKind::CLI) {
+    return false;
+  }
+
+  enum class OptionKind : uint8_t { OptionalValue, RequiredValue, NotAcceptValue };
```

Comment:
It's a bit hard to udnerstand your naming here. What do you think of `enum class option_kind : uint8_t { flag, required, optional }` where:

- `flags` stands for `Individual characters (do not accept values)`
- `required` stands for `Characters followed by a colon (parameter requires value)`
- `optional` stands for `Characters followed by two colons (optional value)`